src/martialarts.cpp

```diff
@@ -840,3 +930,174 @@ float ma_technique::armor_penetration( const player &u, damage_type type ) const
+                buffer << string_format( _( "<header>%s buffs:</header>" ), title );
+                for( const auto &buff : buffs ) {
+                    buffer << std::endl << buff->get_description( passive ) ;
+                }
+                buffer << std::endl << "--" << std::endl;
+            }
+        };
+
+        buff_desc( _( "Passive" ), ma.static_buffs, true );
+        buff_desc( _( "Move" ), ma.onmove_buffs );
+        buff_desc( _( "Hit" ), ma.onhit_buffs );
+        buff_desc( _( "Attack" ), ma.onattack_buffs );
+        buff_desc( _( "Dodge" ), ma.ondodge_buffs );
+        buff_desc( _( "Block" ), ma.onblock_buffs );
+        buff_desc( _( "Get hit" ), ma.ongethit_buffs );
+
+        for( const auto &tech : ma.techniques ) {
+            buffer << string_format( _( "<header>Technique:</header> <bold>%s</bold>   " ), tech.obj().name );
+            buffer << tech.obj().get_description() << std::endl << "--" << std::endl;
+        }
+
+        if( !ma.weapons.empty() ) {
+            buffer << std::endl << std::endl;
+            buffer << ngettext( "<bold>Weapon:</bold>", "<bold>Weapons:</bold>", ma.weapons.size() ) << " ";
+            buffer << enumerate_as_string( ma.weapons.begin(), ma.weapons.end(), []( const std::string & wid ) {
+                return item::nname( wid );
+            } );
+        }
+
+        catacurses::window w = catacurses::newwin( FULL_SCREEN_HEIGHT, FULL_SCREEN_WIDTH,
+                               ( TERMY > FULL_SCREEN_HEIGHT ) ? ( TERMY - FULL_SCREEN_HEIGHT ) / 2 : 0,
+                               ( TERMX > FULL_SCREEN_WIDTH ) ? ( TERMX - FULL_SCREEN_WIDTH ) / 2 : 0 );
+
+        std::string text = replace_colors( buffer.str() );
+        int width = FULL_SCREEN_WIDTH - 4;
+        int height = FULL_SCREEN_HEIGHT - 2;
+        const auto vFolded = foldstring( text, width );
+        int iLines = vFolded.size();
+        int selected = 0;
+
+        input_context ict;
+        ict.register_action( "UP" );
+        ict.register_action( "DOWN" );
+        ict.register_action( "PAGE_UP" );
+        ict.register_action( "PAGE_DOWN" );
+        ict.register_action( "QUIT" );
+
+        do {
+            if( selected < 0 ) {
+                selected = 0;
+            } else if( iLines < height ) {
+                selected = 0;
+            } else if( selected >= iLines - height ) {
+                selected = iLines - height;
+            }
+
+            werase( w );
+            fold_and_print_from( w, 1, 2, width, selected, c_light_gray, text );
+            draw_border( w, BORDER_COLOR, string_format( _( " Style: %s " ), ma.name ) );
+            draw_scrollbar( w, selected, height, iLines, 1, 0, BORDER_COLOR, true );
+            wrefresh( w );
+            catacurses::refresh();
+
+            std::string action = ict.handle_input();
+
+            if( action == "QUIT" ) {
+                break;
+            } else if( action == "DOWN" || action == "PAGE_DOWN" ) {
```

Comment:
Why this? Arrow down and page down are two completely different things, the user is most likely not expecting that.